databricks_cli/unity_catalog/delta_sharing_cli.py

```diff
@@ -369,34 +401,53 @@ def get_provider_cli(api_client, name):
 @click.command(context_settings=CONTEXT_SETTINGS,
                short_help='Update a provider.')
 @click.option('--name', required=True, help='Name of the provider to update.')
-@click.option('--new_name', default=None, help='New name of the provider.')
+@click.option('--new-name', default=None, help='New name of the provider.')
 @click.option('--comment', default=None, required=False,
               help='Free-form text description.')
+@click.option('--owner', default=None, required=False,
+              help='Owner of the provider.')
 @click.option('--recipient-profile-json-file', default=None, type=click.Path(),
               help='File containing recipient profile in JSON format.')
 @click.option('--recipient-profile-json', default=None, type=JsonClickType(),
               help='JSON string containing recipient profile.')
+@click.option('--json-file', default=None, type=click.Path(),
+              help=json_file_help(method='PATCH', path='/providers/{name}'))
+@click.option('--json', default=None, type=JsonClickType(),
+              help=json_string_help(method='PATCH', path='/providers/{name}'))
 @debug_option
 @profile_option
 @eat_exceptions
 @provide_api_client
-def update_provider_cli(api_client, name, new_name, comment, recipient_profile_json_file,
-                        recipient_profile_json):
+def update_provider_cli(api_client, name, new_name, comment, owner, recipient_profile_json_file,
+                        recipient_profile_json, json_file, json):
     """
     Update a provider.
 
     The public specification for the JSON request is in development.
     """
-    if recipient_profile_json is None and recipient_profile_json_file is None:
-        updated_provider = UnityCatalogApi(api_client).update_provider(name, new_name, comment)
-        click.echo(mc_pretty_format(updated_provider))
+    if ((new_name is not None) or (comment is not None) or (owner is not None) or
+        (recipient_profile_json_file is not None) or (recipient_profile_json) is not None):
+        if (json_file is not None) or (json is not None):
+            raise ValueError('Cannot specify JSON if any other update flags are specified')
+        data = {'name': new_name, 'comment': comment, 'owner': owner}
+
+        if (recipient_profile_json_file is None) and (recipient_profile_json is None):
+            provider_json = UnityCatalogApi(api_client).update_provider(name, provider_spec=data)
+            click.echo(mc_pretty_format(provider_json))
+        else:
+            json_cli_base(recipient_profile_json_file, recipient_profile_json,
+                          lambda profile_json: UnityCatalogApi(api_client).update_provider(
+                              name,
+                              provider_spec=merge_dicts(
+                                  data,
+                                  {'recipient_profile_str': mc_pretty_format(profile_json)})),
```

Comment:
I think you can just set the `recipient_profile_str` key prior to this call. No need for a `merge_dicts` call.

Reply:
We can't set this easily without a nested def because it's scoped within a lambda function. With `merge_dicts_shallow`, it can be reused in the future in a easier fashion.